floodlight/io/datasets.py

```diff
@@ -799,3 +801,278 @@ def _download_matches_info(self) -> None:
                 )
                 with open(season_file, "wb") as binary_file:
                     binary_file.write(download_from_url(season_host_url))
+
+
+class IDSSEDataset:
+    """This dataset loads the accompanying data set from the *An integrated dataset of
+     synchronized spatiotemporal and event data in elite soccer* paper. [1]_
+
+    Upon instantiation, the class checks if the specified data already exists in the
+    repository's root ``.data``-folder, and will download the files to this folder if
+    not. The default setting is to load the first match from the dataset. However, any
+    individual match or the entire dataset (~2.4 GB) can be downloaded.
+
+    Parameters
+    ----------
+    dataset_dir_name: str, optional
+        Name of subdirectory where the dataset is stored within the root .data
+        directory. Defaults to 'dfl_dataset'.
+    match_id: str, optional
+        Match-ID of either one of the matches or "all". Defaults to 'J03WMX'. Setting it
+        to one of the matches will download the data of this individual match, if it
+        does not exist in the repository's root ``.data``-folder. Setting it to 'all'
+        will download the data of all matches that do not exist in ``.data``.
+
+    Notes
+    -----
+    The dataset contains seven full matches of raw event and position data for both
+    teams and the ball from the German Men's Bundesliga season 2022/23 first and second
+    division. A detailed description of the dataset as well as the collection process
+    can be found in the accompanying paper. Data for one match can be queried calling
+    the :func:`~DFLDataset.get`-method by specifying the match. The following matches
+    are available::
+
+        matches = {
+        'J03WMX': 1. FC Köln vs. FC Bayern München,
+        'J03WN1': VfL Bochum 1848 vs. Bayer 04 Leverkusen,
+        'J03WPY': Fortuna Düsseldorf vs. 1. FC Nürnberg,
+        'J03WOH': Fortuna Düsseldorf vs. SSV Jahn Regensburg,
+        'J03WQQ': Fortuna Düsseldorf vs. FC St. Pauli,
+        'J03WOY': Fortuna Düsseldorf vs. F.C. Hansa Rostock,
+        'J03WR9': Fortuna Düsseldorf vs. 1. FC Kaiserslautern
+        }
+
+    Examples
+    --------
+    >>> from floodlight.io.datasets import IDSSEDataset
+
+    >>> dataset = IDSSEDataset()
+    # get one sample
+    >>> events_data_objects, position_data_objects = dataset.get()
```

Comment:
``xy_data_objects``?

Reply:
maybe more descriptive if ``.get("J03WMX")``

Reply:
Added the match to the `.get()`

I don't think xy_data_objects fits here. This returns all objects from the `read_position_data_xml()`. So it includes `Teamsheets`, `Pitch` and so on. However as you also commented on the returns below, I will adress it there and adjust here, accordingly